src/tests/ut/reserve_queue/reserve_queue_test.cpp

```diff
@@ -0,0 +1,412 @@
+    sut_.addAwaitingPatient(filling_treatment, patients_[1]);
+    sut_.addAwaitingPatient(filling_treatment, patients_[2]);
+
+    Treatment tooth_extraction_treatment{Treatment::ToothExtraction};
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[3]);
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[0]);
+
+    const auto filling_treatment_patients_before = sut_.awaitingPatients(filling_treatment);
+    const auto tooth_extraction_patients_before = sut_.awaitingPatients(tooth_extraction_treatment);
+
+    ASSERT_THAT(filling_treatment_patients_before, ElementsAre(patients_[0], patients_[1], patients_[2]));
+    ASSERT_THAT(tooth_extraction_patients_before, ElementsAre(patients_[3], patients_[0]));
+
+    sut_.removeAwaitingPatient(filling_treatment, patients_[3]);
+    sut_.removeAwaitingPatient(tooth_extraction_treatment, patients_[1]);
+
+    const auto filling_treatment_patients_after = sut_.awaitingPatients(filling_treatment);
+    const auto tooth_extraction_patients_after = sut_.awaitingPatients(tooth_extraction_treatment);
+
+    EXPECT_EQ(filling_treatment_patients_before, filling_treatment_patients_after);
+    EXPECT_EQ(tooth_extraction_patients_before, tooth_extraction_patients_after);
+}
+
+TEST_F(RemoveAwaitingPatientShould, NotChangeDoctorQueueIfPatientNotPresentInQueue)
+{
+    // first doctor queue
+    sut_.addAwaitingPatient(doctors_[0], patients_[2]);
+    sut_.addAwaitingPatient(doctors_[0], patients_[1]);
+    // second doctor queue
+    sut_.addAwaitingPatient(doctors_[1], patients_[3]);
+    sut_.addAwaitingPatient(doctors_[1], patients_[0]);
+
+    const auto first_doctor_patients_before = sut_.awaitingPatients(doctors_[0]);
+    const auto second_doctor_patients_before = sut_.awaitingPatients(doctors_[1]);
+
+    ASSERT_THAT(first_doctor_patients_before, ElementsAre(patients_[2], patients_[1]));
+    ASSERT_THAT(second_doctor_patients_before, ElementsAre(patients_[3], patients_[0]));
+
+    sut_.removeAwaitingPatient(doctors_[0], patients_[3]);
+    sut_.removeAwaitingPatient(doctors_[1], patients_[1]);
+
+    const auto first_doctor_patients_after = sut_.awaitingPatients(doctors_[0]);
+    const auto second_doctor_patients_after = sut_.awaitingPatients(doctors_[1]);
+
+    EXPECT_EQ(first_doctor_patients_before, first_doctor_patients_after);
+    EXPECT_EQ(second_doctor_patients_before, second_doctor_patients_after);
+}
+
+TEST_F(RemoveAwaitingPatientShould, NotAddTreatmentKeyIfRemovalRequestedFromEmptyList)
+{
+    const auto treatments_awaiting_before = sut_.treatmentsHavingQueue();
+    ASSERT_TRUE(treatments_awaiting_before.empty());
+
+    sut_.removeAwaitingPatient(Treatment::DentalFilling, patients_[0]);
+    sut_.removeAwaitingPatient(Treatment::ToothExtraction, patients_[1]);
+    const auto treatments_awaiting_after_removal_attempts = sut_.treatmentsHavingQueue();
+
+    EXPECT_TRUE(treatments_awaiting_after_removal_attempts.empty());
```

Comment:
Better readibility can be achived by hidding it in method such as 
expectTreatmentsHavingQueue

Reply:
DONE.
I have created a fixture function with proposed name.